interpret.js

```diff
@@ -12,10 +12,10 @@ const find = (xs, f) => {
   return found;
 }
 
-const dispatch = pairs =>
+const dispatch = of => pairs =>
```

Comment:
I generally prefer to curry without trying to think of a use case -- when something is curried, it's trivial to apply it in one go. When it is not, work has to be done to curry it.

In this case, it seems possible (though admittedly not that probable) that someone would want to convert various free monad apps to the same monad, but interpreting the free monads in different ways. For example, you might have an application that should work with various API servers and provide different interpreters for each server